src/rdkafka_msgset_reader.c

```diff
@@ -902,6 +1062,29 @@ rd_kafka_msgset_reader_v2 (rd_kafka_msgset_reader_t *msetr) {
                                               &save_slice, payload_size))
                         rd_kafka_buf_check_len(rkbuf, payload_size);
 
+                if (msetr->msetr_aborted_txns == NULL &&
+                    msetr->msetr_v2_hdr->Attributes & RD_KAFKA_MSGSET_V2_ATTR_CONTROL) {
+                        /* Since there are no aborted transactions, the MessageSet
+                         * must correspond to a commit marker. These are ignored. */
+                        rd_slice_widen(&rkbuf->rkbuf_reader, &save_slice);
+                        goto done;
+                }
+
+                if (msetr->msetr_aborted_txns != NULL &&
+                    msetr->msetr_v2_hdr->Attributes & RD_KAFKA_MSGSET_V2_ATTR_TRANSACTIONAL &&
+                    !(msetr->msetr_v2_hdr->Attributes & RD_KAFKA_MSGSET_V2_ATTR_CONTROL)) {
+
+                        int64_t txn_start_offset = rd_kafka_aborted_txns_next_aborted_txn_offset_for_pid(
+                                msetr->msetr_aborted_txns, msetr->msetr_v2_hdr->PID);
+                        
+                        if (txn_start_offset >= 0 &&
```

Comment:
the contract with for_pid is -1 on error, so use that here:
`if (txn_start_offset != -1 && `

Reply:
don't necessarily agree it's better but ok.

Reply:
it is a nit, do as you please :)